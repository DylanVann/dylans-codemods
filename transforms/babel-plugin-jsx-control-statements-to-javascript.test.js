import { defineSnapshotTest } from 'jscodeshift/dist/testUtils'
import transform from './babel-plugin-jsx-control-statements-to-javascript'

const transformOptions = {}

defineSnapshotTest(
  transform,
  transformOptions,
  `
function MyComp() {
  return (
    <If condition={true}>
      one
      {'two'}
      <span>three</span>
      <span>four</span>
    </If>
  )
}
`,
  'outermost element of return statement',
)

defineSnapshotTest(
  transform,
  transformOptions,
  `
function MyComp() {
  return (
    <span>
      <If condition={true}>
        one
        {'two'}
        <span>three</span>
        <span>four</span>
      </If>
    </span>
  )
}
`,
  'inside jsx',
)

defineSnapshotTest(
  transform,
  transformOptions,
  `
function MyComp() {
  return (
    <span>
      <If condition={true}>
        <span>four</span>
      </If>
    </span>
  )
}
`,
  'single child',
)

defineSnapshotTest(
  transform,
  transformOptions,
  `
function MyComp() {
  return <If condition={thing && otherThing}>text</If>
}
`,
  'just text children',
)

defineSnapshotTest(
  transform,
  transformOptions,
  `
function MyComp() {
  return (
    <Choose>
      <When condition={test1}>
        <span>IfBlock1</span>
      </When>
      <When condition={test2}>
        <span>IfBlock2</span>
      </When>
      <Otherwise>
        <span>ElseBlock</span>
      </Otherwise>
    </Choose>
  )
}
`,
  'choose',
)
