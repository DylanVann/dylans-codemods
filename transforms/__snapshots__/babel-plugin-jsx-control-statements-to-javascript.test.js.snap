// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inside jsx 1`] = `
"function MyComp() {
  return (
    <span>
      {Boolean(true) && <React.Fragment>one{'two'}
        <span>three</span>
        <span>four</span>
      </React.Fragment>}
    </span>
  );
}"
`;

exports[`just test 1`] = `
"function MyComp() {
  return Boolean(thing && otherThing) && <React.Fragment>text</React.Fragment>;
}"
`;

exports[`outermost element of return statement 1`] = `
"function MyComp() {
  return Boolean(true) && <React.Fragment>one{'two'}
    <span>three</span>
    <span>four</span>
  </React.Fragment>;
}"
`;

exports[`single child 1`] = `
"function MyComp() {
  return (
    <span>
      {Boolean(true) && <span>four</span>}
    </span>
  );
}"
`;
